src/navsat_transform.cpp

```diff
@@ -110,66 +118,68 @@ void NavSatTransform::run()
       datum_yaw = datum_vals[2];
     }
 
-    robot_localization::srv::set_datum::request request;
-    request.geo_pose.position.latitude = datum_lat;
-    request.geo_pose.position.longitude = datum_lon;
-    request.geo_pose.position.altitude = 0.0;
+    auto request = std::make_shared<robot_localization::srv::SetDatum::Request>();
+    request->geo_pose.position.latitude = datum_lat;
+    request->geo_pose.position.longitude = datum_lon;
+    request->geo_pose.position.altitude = 0.0;
     tf2::Quaternion quat;
     quat.setRPY(0.0, 0.0, datum_yaw);
-    request.geo_pose.orientation = tf2::toMsg(quat);
-    robot_localization::srv::set_datum::response response;
+    request->geo_pose.orientation = tf2::toMsg(quat);
+    auto response = std::make_shared<robot_localization::srv::SetDatum::Response>();
     datumCallback(request, response);
   }
 
-  auto odom_sub = node_->create_subscription<nav_msgs::msg::Odometry>(
-    "odometry/filtered", std::bind(&NavSatTransform::odomCallback, this, _1));
-  auto gps_sub = node_->create_subscription<sensor_msgs::msg::NavSatFix>(
-    "gps/fix", std::bind(&NavSatTransform::gpsFixCallback, this, _1));
+  odom_sub = this->create_subscription<nav_msgs::msg::Odometry>(
+    "odometry/filtered", 10, std::bind(&NavSatTransform::odomCallback, this, _1));
+
+  gps_sub = this->create_subscription<sensor_msgs::msg::NavSatFix>(
+    "gps/fix", 10, std::bind(&NavSatTransform::gpsFixCallback, this, _1));
 
   if (!use_odometry_yaw_ && !use_manual_datum_) {
-    auto imu_sub = node_->create_subscription<sensor_msgs::msg::Imu>(
-      "imu", std::bind(&NavSatTransform::imuCallback, this, _1));
+    imu_sub = this->create_subscription<sensor_msgs::msg::Imu>(
+      "imu", 10, std::bind(&NavSatTransform::imuCallback, this, _1));
   }
 
-  auto gps_odom_pub =
-    node_->create_publisher<nav_msgs::msg::Odometry>("odometry/gps");
-  rclcpp::Subscription::SharedPtr filtered_gps_pub;
+  gps_odom_pub =
+    this->create_publisher<nav_msgs::msg::Odometry>("odometry/gps", rclcpp::QoS(10));
 
   if (publish_gps_) {
     filtered_gps_pub =
-      node_->create_publisher<sensor_msgs::msg::NavSatFix>("gps/filtered");
+      this->create_publisher<sensor_msgs::msg::NavSatFix>("gps/filtered", rclcpp::QoS(10));
   }
 
   // Sleep for the parameterized amount of time, to give
   // other nodes time to start up (not always necessary)
-  rclcpp::utilities::sleep_for(std::chrono::seconds(delay));
+  rclcpp::sleep_for(std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::duration<double>(delay)));
```

Comment:
You changed this from `seconds` to `nanoseconds` for the unit on `delay`, is that intentional? I think seconds was the more logical unit. Open for comments